pkg/network/namescheme/networknamescheme_test.go

```diff
@@ -23,42 +23,254 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
 	virtv1 "kubevirt.io/api/core/v1"
 
 	"kubevirt.io/kubevirt/pkg/network/namescheme"
 )
 
 var _ = Describe("Network Name Scheme", func() {
-	Context("CreateNetworkNameScheme", func() {
-		DescribeTable("should return the expected NetworkNameSchemeMap",
-			func(networkList []virtv1.Network, expectedNetworkNameSchemeMap map[string]string) {
-				podIfacesNameScheme := namescheme.CreateNetworkNameScheme(networkList)
+	DescribeTable("create pod interfaces name scheme",
+		func(nameSchemeFn func([]virtv1.Network) map[string]string, networkList []virtv1.Network, expectedNetworkNameScheme map[string]string) {
```

Comment:
can you export `func([]virtv1.Network) map[string]string` into a type ? I had to re-read this functions signature 3 times until I understood what was happening.